lib/bridge.js

```diff
@@ -237,11 +240,109 @@ Bridge.prototype.run = function(port, config, appServiceInstance) {
         }
         self.opts.controller.onLog(line, false);
     });
+    this.customiseAppservice(this.appService);
     this.appService.listen(port);
     return this.loadDatabases();
 };
 
 /**
+ * Apply any customisations required on the appService object.
+ * Subclasses are intended to override this method where appropriate. If
+ * overridden the subclass method will still have to call the one in this
+ * class.
+ * @param {AppService} the AppService instance of the object
+ */
+Bridge.prototype.customiseAppservice = function(appService) {
+    // TODO(paul): This is gut-wrenching into the AppService instance itself.
+    //   Maybe an API on that object would be good?
+    var app = appService.app;
+
+    if (this.opts.controller.thirdPartyLookup) {
+        var lookupController = this.opts.controller.thirdPartyLookup;
+
+        var protocolFilter;
+        if ("protocols" in lookupController) {
+            protocolFilter = {};
+            lookupController.protocols.forEach(function (p) {
+                protocolFilter[p] = true;
+            });
+        }
+
+        app.get("/_matrix/app/:ver/thirdparty/protocol/:protocol", (req, res) => {
+            if (req.params.ver != "unstable") {
+                res.status(404).send("Cannot GET " + req.path + "\r\n");
+                return;
+            }
+
+            if (!lookupController.getProtocol) {
+                res.status(404).send("Cannot GET " + req.path + "\r\n");
+                return;
+            }
+
+            var protocol = req.params.protocol;
+
+            if (protocolFilter && !(protocol in protocolFilter)) {
+                res.status(404).json({err: "Unknown 3PN protocol " + protocol});
+                return;
+            }
+
+            lookupController.getProtocol(protocol).then(
+                (result) => { res.status(200).json(result) },
+                (e)      => { res.status(e.code).json({error: e.err}) }
```

Comment:
No ES6 fat arrow functions please - ES5 only! Also, Please stick to existing whitespace styles: so no alignment on functions like this please (I admit it looks nice though :)).